pkg/controllers/pod/pod_controller.go

```diff
@@ -303,9 +317,11 @@ func (c *PodController) populateEndpointCache() error {
 		return err
 	}
 
+	c.endpointCache.Lock()
 	for _, wep := range workloadEndpointList.Items {
-		c.endpointCache[wep.Metadata.Workload] = wep
+		c.endpointCache.m[wep.Metadata.Workload] = wep
 	}
+	c.endpointCache.Unlock()
```

Comment:
It's better if you perform the Unlock in a `defer` statement so there's no chance of it accidentally getting held.

e.g.

```go
c.endpointCache.Lock()
defer c.endpointCache.Unlock()
. . .
```

Reply:
I suppose in this case it's pretty low risk so this is fine.